server/controllers/finance/reports/incomeExpense/index.js

```diff
@@ -0,0 +1,283 @@
+        WHERE posting_journal.account_id=? AND (posting_journal.trans_date >=? AND posting_journal.trans_date <=?)
+      ) UNION (
+      SELECT general_ledger.project_id, general_ledger.uuid, general_ledger.trans_date, general_ledger.debit_equiv, general_ledger.credit_equiv,
+        general_ledger.debit, general_ledger.credit, general_ledger.account_id, general_ledger.record_uuid, general_ledger.entity_uuid,  general_ledger.reference_uuid,
+        general_ledger.currency_id, general_ledger.trans_id, general_ledger.description, general_ledger.comment, general_ledger.origin_id, general_ledger.user_id 
+      FROM general_ledger
+        WHERE general_ledger.account_id=? AND (general_ledger.trans_date >=? AND general_ledger.trans_date <=?)
+      )   
+    ) AS t  
+    JOIN user u ON t.user_id = u.id
+    JOIN account a ON t.account_id = a.id
+    LEFT JOIN transaction_type tr ON tr.id = t.origin_id
+    WHERE t.debit > 0 GROUP BY t.trans_id;`;
+  return db.exec(query, [accountId, dateFrom, dateTo, accountId, dateFrom, dateTo]);
+}
+
+
+/**
+ * @function getExpenseReport
+ * @param {account} account id of cashbox
+ * @param {date} dateFrom A starting date
+ * @param {date} dateTo A stop date
+ */
+function getExpenseReport(accountId, dateFrom, dateTo) {
+  let query =`
+    SELECT t.project_id, BUID(t.uuid) AS uuid, t.trans_date, t.debit_equiv, t.credit_equiv, t.debit, t.credit, t.account_id, BUID(t.record_uuid) AS record_uuid,
+    BUID(t.entity_uuid) AS entity_uuid,  BUID(t.reference_uuid) AS record_uuid, t.currency_id, t.trans_id, t.description, t.comment, t.origin_id, t.user_id, u.username, 
+    a.number, tr.text AS transactionType, a.label
+    FROM 
+    (
+      (
+      SELECT posting_journal.project_id, posting_journal.uuid, posting_journal.trans_date, posting_journal.debit_equiv, posting_journal.credit_equiv,
+        posting_journal.debit, posting_journal.credit, posting_journal.account_id, posting_journal.record_uuid, posting_journal.entity_uuid,  posting_journal.reference_uuid,
+        posting_journal.currency_id, posting_journal.trans_id, posting_journal.description, posting_journal.comment, posting_journal.origin_id, posting_journal.user_id 
+      FROM posting_journal
+        WHERE posting_journal.account_id=? AND (posting_journal.trans_date >=? AND posting_journal.trans_date <=?)
+      ) UNION (
+      SELECT general_ledger.project_id, general_ledger.uuid, general_ledger.trans_date, general_ledger.debit_equiv, general_ledger.credit_equiv,
+        general_ledger.debit, general_ledger.credit, general_ledger.account_id, general_ledger.record_uuid, general_ledger.entity_uuid,  general_ledger.reference_uuid,
+        general_ledger.currency_id, general_ledger.trans_id, general_ledger.description, general_ledger.comment, general_ledger.origin_id, general_ledger.user_id 
+      FROM general_ledger
+        WHERE general_ledger.account_id=? AND (general_ledger.trans_date >=? AND general_ledger.trans_date <=?)
+      )   
```

Comment:
As you suppose it, we must be careful when comparing dates without using MySQL DATE function.